hdr/sqlite_modern_cpp.h

```diff
@@ -340,6 +340,34 @@ namespace sqlite {
 			return sqlite3_last_insert_rowid(_db.get());
 		}
 
+#ifdef SQLITE_HAS_CODEC
+		database(const std::string &db_name, const std::string &key): database(db_name) {
+			set_key(key);
+		}
+		database(const std::u16string &db_name, const std::string &key): database(db_name) {
+			set_key(key);
+		}
+
+		void set_key(const std::string &key) {
+			if(auto ret = sqlite3_key(_db.get(), key.data(), key.size()))
+				exceptions::throw_sqlite_error(ret);
```

Comment:
There is no such thing.
SQLCipher isn't very good at error reporting.
For example if the key is wrong, it still returns SQLITE_OK and fails only if you try to access the database for the first time with the existing SQLITE_NOTADB.
The key and rekey functions only fail, if you pass them nullpointers, but this can't happen here.
So actually this are noops, but the functions are a part of sqlites more generic codec interface and other implementations may choose to fail here, so we should at least check the return code.